android-components/components/browser/toolbar/src/main/res/layout/mozac_browser_toolbar_displaytoolbar.xml

```diff
@@ -105,6 +105,7 @@
         android:layout_height="40dp"
         android:layout_marginTop="8dp"
         android:paddingEnd="@dimen/mozac_browser_toolbar_origin_padding_end"
+        android:accessibilityLiveRegion="polite"
```

Comment:
The previous implementation with `announceForAccessibility` was not the recommended one and thanks to @mcarare, who recommended me to use accessibility live region, I achieved a similar results but with a cleaner implementation. 
Video (there is no audio in the recording for some reason...):

https://user-images.githubusercontent.com/32488956/219055185-564fe7b8-9c7e-47e7-bb85-16d74f5e1679.mp4

Reply:
With this approach we are reading the full url + "Search or enter address" as an action but without the user imediately being able to - other Views may be selected.
With the previous approach we would be reading the page title, the same as it happens on Chrome.
A bigger issue though is that with this approach everytime the page is scrolled the announcement is repeated. On a Samsung A53 with Android 12 I saw the same announcement repeated every ~20 seconds if leaving the device alone.

(the displayed text is what Talkback would read)

https://user-images.githubusercontent.com/11428869/219067330-a17212f0-e153-410b-881e-04868217c984.mp4

Reply:
Indeed, it seems like this approach would not fit in this case. I have reverted back to the previous one.